### DISCUSSION TITLE
Channel Plugin Based on Existing Media Files

### DISCUSSION THREAD

**Discussion by anu1097** [jellyfin/jellyfin-meta#39] [Ideas]
Hi, 

I want to make a feature and add to this repo, that allows to create local channels out of existing media files. 
Take movies for example, movies can be categories into Genres. Each Genre could be a channel. 

Primary Motivation is to remove the time taken to select a movie/media file. 

Initial Feature List I have in mind -
1. Create Channels Based on Genre
2. Randomise movies based on Days, create schedule
3. Show Schedule.
4. Show other movie trailers in between movie playback, users shall be given an option to select how many ads. Typically two ads, so movie is splitted in 3 sections. Instead of Trailers, it can be teasers too.
5. Allow User to skip to next movie in initial 10 min.
6. 1 Custom Channel, to let user customise the schedule.

For TV I haven't thought too much at the moment. But it could be similar, in future I plan to create different kinds of channels based on common features of Movies or TV Shows.

**Comment by jedthe3rd** [comment]
If this ends up being something that doesn't fit in jellyfin functionality you can use this:
https://ersatztv.org/

**Comment by anu1097** [comment]
@jedthe3rd looks interesting, but I am really used to Jellyfin's UI and various client apps. Don't want to switch to another medium. Willing to invest coding efforts here. But I saw this in their Contribution Guideline to first post an Enhancement request here before building a big feature.

EDIT: It seems Jellyfin is supported https://ersatztv.org/user-guide/configure-clients/

I will check this out.

**Comment by thornbill** [comment]
Feature requests belong on features.jellyfin.org. Here is one that seems related: https://features.jellyfin.org/posts/812/create-live-tv-channels-streaming-your-own-media

**Comment by anu1097** [comment]
No this wasn't a feature request, I read somewhere in the Readme to open an issue if I have a large Pull Request in Mind and discuss technical implementation prior. 
I was thinking of taking the route of creating a plugin, wanted to discuss that first.

**Comment by thornbill** [comment]
Ah I see. A plugin would belong in a separate git repository and would not be added via a pull request here, but I understand where you are coming from now. I think the best place to discuss how to get started would be in the development channel on matrix or discord. Information on how to join can be found [on the website](https://jellyfin.org/contact/).

**Comment by nielsvanvelzen** [comment]
A limitation of Jellyfin plugins is that they aren't flexible at all in their user interface. You can provide pages for the dashboard and that's it. Some third-party plugins hack around this by injecting JavaScript but that's not something we want to support. So either the plugin has it's completely separate frontend or it needs to be integrated in the official clients (at which point it can't be a plugin anymore).

So the real question here is whether to integrate such a feature into Jellyfin or not. I'll move the issue to our "meta" repository which is used to discuss these kinds of things.

Personally I think it would be a nice addition, if kept simple. For more advanced use there are already great third-party tools like Ersatz.

**Reply by thornbill** [reply]
While this would be kind of a neat feature, ersatz seems like it already fits this need. We do have a [feature request](https://features.jellyfin.org/posts/812/create-live-tv-channels-streaming-your-own-media) open for this type of functionality, but it's really not all that popular. Maybe a _very_ basic implementation could be done as a plugin and we could get a better sense of demand from there? But as you mentioned the UI would be a challenge to do in a plugin even just covering the basics.

Also some of the functionality listed doesn't seem like it would be worthwhile because every client would require a unique implementation and no other live TV source is going to support things like "Allow User to skip to next movie in initial 10 min".

**Reply by net-shaper** [reply]
I also support this idea. Perhaps it could be a new type of _TV Guide Provider_ that is self-referential to the current Jellyfin instance instead of an external provider. That way the TV Guide UI could be re-used. 

The _ersatz_ approach seems a little cyclic to me; Ersatz takes the content from my Jellyfin library and streams it back to my Jellyfin server which streams it to my Jellyfin client(s).
